File: solutions/add-digits.cpp
/*
 *Solution of problem 258
 *https://leetcode.com/problems/add-digits/
 */

class Solution {
public:
    int addDigits(int num) {
        return (1 + (num - 1) % 9);
    }
};

/*
 *    int addDigits(int num) {
        int ans = 9;
        if(num == 0){
            ans = 0;
            return ans;
        }
        if(num%9 == 0 && num>0)
            return ans;
        else
            return(num%9);
    }
 *
 *More info: https://en.wikipedia.org/wiki/Digital_root
 * /

File: solutions/add-digits.py
"""
258. Add Digits
https://leetcode.com/problems/add-digits/
"""
class Solution:
    def addDigits(self, num: int) -> int:

        # to solve this problem in O(1)
        # there is property of numbers
        # if the number is divided by 9 then its sum of digits is also divided by 9

        # base case
        if num is 0:
            return 0

        # if num is divided by 9 return 9
        if num % 9 == 0:
            return 9

        # for all other numbers
        else:
            return num % 9
